plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py

```diff
@@ -34,13 +34,16 @@ def manipulate_search_path(self, search_path: ConfigSearchPath) -> None:
         )
 
 
-class SubmititLauncher(Launcher):
+class LocalSubmititLauncher(Launcher):
+
+    _EXECUTOR = "local"
+
     def __init__(self, **params: Any) -> None:
         self.params = OmegaConf.structured(
-            LocalParams if params["executor"].value == "local" else SlurmParams
+            LocalParams if self._EXECUTOR == "local" else SlurmParams
         )
         for key, val in params.items():
-            OmegaConf.update(self.params, key, val if key != "executor" else val.value)
+            OmegaConf.update(self.params, key, val)
```

Comment:
I am surprised that worked in the previous version when you use the actual object and not a DictConfig.

regardless, you can get rid of this loop with the example I gave you above.